### PR TITLE
Update BPKText to use LocalizedStringKey for text

### PR DESCRIPTION
**Pull request by frugoman**
This pull request updates the BPKText struct to use LocalizedStringKey for the text parameter instead of a plain string. It also adds a test for styled text in BPKText.

image

### CODE REVIEW

File: Backpack-SwiftUI/Tests/Text/BPKTextTests.swift [hunk 1]
```diff
@@ -40,10 +40,14 @@ class BPKTextTests: XCTestCase {
...
     
+    func test_withStyledText() {
+        assertSnapshot(BPKText("Hello **Backpack**, I hope you're having a _great_ day!", style: .heading1))
+    }
+    
     func test_customColor() {
         assertSnapshot(
             BPKText("Hello Backpack, I hope you're having a great day!")
-                .foregroundColor(.primaryColor)
+                .foregroundColor(.coreAccentColor)
```
Comment (frugoman) [inline Backpack-SwiftUI/Tests/Text/BPKTextTests.swift:50]: deprecated colour!

File: Backpack-SwiftUI/Tests/Text/BPKTextTests.swift [hunk 2]
```diff
@@ -40,10 +40,14 @@ class BPKTextTests: XCTestCase {
         assertSnapshot(BPKText("Hello Backpack, I hope you're having a great day!", style: .heading1))
     }
     
+    func test_withStyledText() {
+        assertSnapshot(BPKText("Hello **Backpack**, I hope you're having a _great_ day!", style: .heading1))
```
Comment (gert-janvercauteren) [inline Backpack-SwiftUI/Tests/Text/BPKTextTests.swift:44]: Does this markdown formatting translate to iOS 14 as well?

Comment (frugoman) [reply Backpack-SwiftUI/Tests/Text/BPKTextTests.swift:44]: It does not, so It'd be up for the consumer in this case to determine if they should or should not use it.
Although we're getting rid of 14 now, so it does not worry me a whole lot! but up for debate!

File: Backpack-SwiftUI/Text/Classes/BPKText.swift [hunk 3]
```diff
@@ -25,15 +25,15 @@ import SwiftUI
...
+    private var text: LocalizedStringKey
     private var style: BPKFontStyle
     
     private var textColor = Color(BPKColor.textPrimaryColor)
     private var lineLimit: Int? = 1
     private var strikethrough: Bool = false
     
     public init(_ text: String, style: BPKFontStyle = .bodyDefault) {
-        self.text = text
+        self.text = .init(text)
```
Comment (gert-janvercauteren) [inline Backpack-SwiftUI/Text/Classes/BPKText.swift:36]: Can we use the other LocalizedStringKey(text) init to make it clearer?

**Review by gert-janvercauteren — APPROVED** [review]

**Pull request merged**